src/App.tsx

```diff
@@ -16,35 +18,74 @@
 ];
 
 export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button type="button" className="button is-info is-light">
-          Sort alphabetically
-        </button>
+  const [sortField, setsortField] = useState('');
+  const [isReversed, setReverse] = useState(false);
 
-        <button type="button" className="button is-success is-light">
-          Sort by length
-        </button>
+  const SORT_ALPHABETICALLY = 'Sort alphabetically';
+  const SORT_LENGTH = 'Sort by length';
 
-        <button type="button" className="button is-warning is-light">
-          Reverse
-        </button>
+  let visibleGoods = [...goodsFromServer].sort((good1, good2) => {
+    switch (sortField) {
+      case SORT_ALPHABETICALLY:
+        return good1.localeCompare(good2);
+      case SORT_LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
 
-        <button type="button" className="button is-danger is-light">
-          Reset
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  const reset = () => {
+    visibleGoods = goodsFromServer;
```

Comment:
Assigning `goodsFromServer` directly to `visibleGoods` will not trigger a re-render. Consider using a state variable for `visibleGoods` and updating it with a setter function to ensure the UI updates correctly.